### PR TITLE
 use latest clusterimageset version

### PR DESCRIPTION
**Pull request by ldpliu**
Signed-off-by: ldpliu <daliu@redhat.com>
https://github.com/open-cluster-management/backlog/issues/18331
https://github.com/open-cluster-management/backlog/issues/18339

### CODE REVIEW

File: pkg/utils/cluster.go [hunk 1]
```diff
@@ -343,6 +343,41 @@ with image %s ===============================`, clusterName, imageRefName)
 
 }
 
+// the string format should be like: img4.8.10-x86_64.yaml
+// if s1> s2, return true, or return false
+func compareImageVersion(s1, s2 string) (bool, error) {
+	s1VersionLastIndex := strings.Index(s1, "-")
+	s1VersionStr := s1[3:s1VersionLastIndex]
```
Comment (qiujian16) [inline pkg/utils/cluster.go:350]: Is it safe to do that with out length check?

File: pkg/utils/cluster.go [hunk 2]
```diff
@@ -343,6 +343,41 @@ with image %s ===============================`, clusterName, imageRefName)
...
+
+	s2VersionLastIndex := strings.Index(s2, "-")
+	s2VersionStr := s2[3:s2VersionLastIndex]
+
+	s1VersionArray := strings.Split(s1VersionStr, ".")
+	s2VersionArray := strings.Split(s2VersionStr, ".")
+
+	for i, s1 := range s1VersionArray {
+		if len(s2VersionArray) <= i {
```
Comment (qiujian16) [inline pkg/utils/cluster.go:359]: should you check if the two array has the same length upfront?

**Review by qiujian16 — COMMENTED** [review]
can we use https://github.com/blang/semver to parseVersion?

**Comment by skeeey** [comment]
Use new pr #61, closed

**Pull request closed without merging**